catalyst-gateway/bin/src/service/api/documents/post_document_index_query/mod.rs

```diff
@@ -103,14 +98,25 @@ pub(crate) async fn endpoint(
 /// Fetch documents from the event db
 async fn fetch_docs(
     conditions: &DocsQueryFilter, query_limits: &QueryLimits,
-) -> anyhow::Result<Vec<IndexedDocumentDocumented>> {
+) -> anyhow::Result<(Vec<IndexedDocumentDocumented>, u32)> {
     let docs_stream = SignedDocBody::retrieve(conditions, query_limits).await?;
     let indexed_docs = DashMap::new();
 
+    let mut total_fetched_doc_count: u32 = 0;
     docs_stream
         .try_for_each(|doc: SignedDocBody| {
             let id = *doc.id();
             indexed_docs.entry(id).or_insert_with(Vec::new).push(doc);
+            match total_fetched_doc_count.checked_add(1) {
```

Comment:
What do you think of changing this `try_for_each` to 
```rust
let (indexed_docs, total_fetched_doc_count) = docs_stream .try_fold(
(HashMap::new(), 0),
 |(mut indexed_docs, mut total_fetched_doc_count), doc| 
 async move {
       let id = *doc.id();
       indexed_docs.entry(id).or_insert_with(Vec::new).push(doc);
       total_fetched_doc_count.checked_add(1).ok_or(anyhow::anyhow!(
                        "Fetched Signed Documents overflow"
                    ))?;
  Ok(indexed_docs, total_fetched_doc_count)     
 }
 ).await?;
 ```
 If I am not mistaken it could allow to avoid using of `DashMap` and use simple `HashMap`

Reply:
Yup, this works.